include/aos/common/tools/memory.hpp

```diff
@@ -12,6 +12,59 @@
 
 namespace aos {
 
+/**
+ * Default deleter invokes delete operator for the given pointer.
+ */
+template <typename T>
+class DefaultDeleter : public NonCopyable {
+public:
+    /**
+     * Constructor that takes a pointer to allocator.
+     *
+     * @param allocator input allocator.
+     */
+    DefaultDeleter(Allocator* allocator = nullptr)
+        : mAllocator(allocator)
+    {
+    }
+
+    /**
+     * Move constructor for a DefaultDeleter of a derived class.
```

Comment:
Moves

Reply:
Basically move is not a verb  but noun, like "copy constructor"